estore_project/users/views.py

```diff
@@ -1,45 +1,60 @@
 from django.contrib.auth import get_user_model
-from django.contrib.auth.mixins import LoginRequiredMixin
-from django.contrib.messages.views import SuccessMessageMixin
-from django.urls import reverse
-from django.utils.translation import gettext_lazy as _
-from django.views.generic import DetailView, RedirectView, UpdateView
+from django.contrib.auth.hashers import check_password
+from rest_framework import status
+from rest_framework.generics import GenericAPIView
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+from rest_framework_simplejwt.tokens import RefreshToken
 
-User = get_user_model()
-
-
-class UserDetailView(LoginRequiredMixin, DetailView):
-
-    model = User
-    slug_field = "username"
-    slug_url_kwarg = "username"
-
-
-user_detail_view = UserDetailView.as_view()
-
-
-class UserUpdateView(LoginRequiredMixin, SuccessMessageMixin, UpdateView):
-
-    model = User
-    fields = ["name"]
-    success_message = _("Information successfully updated")
+from .serializers import UserSerializer
 
-    def get_success_url(self):
-        return self.request.user.get_absolute_url()  # type: ignore [union-attr]
-
-    def get_object(self):
-        return self.request.user
-
-
-user_update_view = UserUpdateView.as_view()
-
-
-class UserRedirectView(LoginRequiredMixin, RedirectView):
-
-    permanent = False
-
-    def get_redirect_url(self):
-        return reverse("users:detail", kwargs={"username": self.request.user.username})
+User = get_user_model()
 
 
-user_redirect_view = UserRedirectView.as_view()
+class Signup(GenericAPIView):
+    permission_classes = (AllowAny,)
+    serializer_class = UserSerializer
+    queryset = ""
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=self.request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+
+        user = User.objects.get(username=self.request.data["username"])
+
+        refresh = RefreshToken.for_user(user)
+        res = {
```

Comment:
why don't we look for a package that would take care of creating tokens for us

Reply:
`simplejwt` provides a class based view `TokenObtainPairView` that takes care of doing the authentication, creating and returning the toekns

Reply:
Okay